src/databricks/labs/dqx/check_funcs.py

```diff
@@ -2705,3 +2784,53 @@ def _validate_sql_query_params(query: str, merge_columns: list[str]) -> None:
         raise UnsafeSqlQueryError(
             "Provided SQL query is not safe for execution. Please ensure it does not contain any unsafe operations."
         )
+
+
+def _calculate_median_absolute_deviation(df: DataFrame, column: str) -> tuple[float, float]:
+    """
+    Calculate the Median Absolute Deviation (MAD) for a numeric column.
+
+    The MAD is a robust measure of variability based on the median, calculated as:
+    MAD = median(|X_i - median(X)|)
+
+    This is useful for outlier detection as it is more robust to outliers than
+    standard deviation.
+
+    Args:
+        df: PySpark DataFrame
+        column: Name of the numeric column to calculate MAD for
+
+    Returns:
+        float: The Median Absolute Deviation value
+
+    Raises:
+        ValueError: If column doesn't exist or contains no numeric values
+
+    Example:
+        >>> df = spark.createDataFrame([(1.0,), (2.0,), (3.0,), (4.0,), (5.0,)], ["values"])
+        >>> mad = calculate_median_absolute_deviation(df, "values")
+        >>> print(f"MAD: {mad}")
+    """
+    # Validate column exists
+    if column not in df.columns:
+        raise ValueError(f"Column '{column}' does not exist in DataFrame")
+
+    # Remove null values
+    df_clean = df.filter(F.col(column).isNotNull())
+
+    if df_clean.count() == 0:
+        raise ValueError(f"Column '{column}' contains no numeric values")
```

Comment:
this should not result in failure, but condition evaluation to False, so that the check does not fail. It could be that all rows are null which is a valid scenario. Also data type check has been done before in the outer function.